projects/timer_challenge/src/main.c

```diff
@@ -0,0 +1,42 @@
+#include "interrupt.h"   // interrupts are required for soft timers
+#include "log.h"         // for printing
+#include "soft_timer.h"  // for soft timers
+#include "wait.h"        // for wait function
+
+#define COUNTER_PERIOD 500
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+void prv_timer_callback(SoftTimerId timer_id, void *context) {
```

Comment:
nit: private functions should actually be marked `static` to ensure they're not called elsewhere